connector/countconnector/config_integration_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build integration
+
+package countconnector
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/connector/connectortest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/otelcol/otelcoltest"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/connector/countconnector/internal/metadata"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/pmetrictest"
+)
+
+func getValidatedCountConfigFromYAML(t *testing.T, configPath string) (*Config, error) {
+	t.Helper()
+
+	factories, err := otelcoltest.NopFactories()
+	if err != nil {
+		return nil, fmt.Errorf("create nop factories: %w", err)
+	}
+	factories.Connectors[metadata.Type] = NewFactory()
+
+	cfg, err := otelcoltest.LoadConfigAndValidate(configPath, factories)
+	if err != nil {
+		return nil, fmt.Errorf("load and validate config %q: %w", configPath, err)
+	}
+
+	connectorID := component.NewID(metadata.Type)
+	ccfg, ok := cfg.Connectors[connectorID].(*Config)
+	if !ok {
+		return nil, fmt.Errorf("connector %q config is not of type *Config", connectorID)
+	}
+	if ccfg == nil {
+		return nil, fmt.Errorf("connector %q config is nil", connectorID)
+	}
+
+	return ccfg, nil
+}
+
+func runLogsToMetrics(t *testing.T, ccfg *Config, logs plog.Logs) pmetric.Metrics {
+	t.Helper()
+
+	sink := new(consumertest.MetricsSink)
+	conn, err := NewFactory().CreateLogsToMetrics(t.Context(), connectortest.NewNopSettings(metadata.Type), ccfg, sink)
+	require.NoError(t, err)
+	require.NoError(t, conn.Start(t.Context(), componenttest.NewNopHost()))
+	t.Cleanup(func() { assert.NoError(t, conn.Shutdown(context.Background())) }) //nolint:usetesting
+
+	require.NoError(t, conn.ConsumeLogs(t.Context(), logs))
+	all := sink.AllMetrics()
+	require.Len(t, all, 1)
+	return all[0]
+}
+
+// Test that empty subconfigs use defaults.
+func TestCount_EmptySubconfig_ProducesDefaultLogRecordCount(t *testing.T) {
+	tests := []struct {
+		name       string
+		configFile string
+	}{
+		{
+			name:       "empty top-level count config",
+			configFile: "config-count-empty.yaml",
+		},
+		{
+			name:       "empty logs section",
+			configFile: "config-count-logs-empty.yaml",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ccfg, err := getValidatedCountConfigFromYAML(t, filepath.Join("testdata", tt.configFile))
```

Comment:
I think it'd be best after each call to `getValidatedCountConfigFromYAML` that we add a check `require.NotNil` on the returned `ccfg`. This is unnecessary with existing logic, but it'll potentially help protect us from failures with future changes to functionality.